### ISSUE TITLE
Error unmarshalling content: invalid character '<' looking for beginning of value.

### ISSUE THREAD

**Issue by Wordsgames** [moby/moby#40419]
**Description**

I try to create a docker container from an image from docker hub.

**Steps to reproduce the issue:**
1. Docker run hello-world

**Describe the results you received:**

I would expect docker to pull the image and create a container

**Describe the results you expected:**

I got action and error messages :

Unable to find image 'hello-world:latest' locally
Trying to pull repository registry.access.redhat.com/hello-world ...
Pulling repository registry.access.redhat.com/hello-world
Trying to pull repository docker.io/library/hello-world ...
Trying to pull repository registry.fedoraproject.org/hello-world ...
Pulling repository registry.fedoraproject.org/hello-world
Trying to pull repository quay.io/hello-world ...
Pulling repository quay.io/hello-world
Trying to pull repository registry.centos.org/hello-world ...
Pulling repository registry.centos.org/hello-world
Trying to pull repository docker.io/library/hello-world ...
/usr/bin/docker-current: error unmarshalling content: invalid character '<' looking for beginning of value.
See '/usr/bin/docker-current run --help'.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:         1.13.1
 API version:     1.26
 Package version: docker-1.13.1-108.git4ef4b30.el7.x86_64
 Go version:      go1.10.3
 Git commit:      4ef4b30/1.13.1
 Built:           Fri Dec 13 01:48:25 2019
 OS/Arch:         linux/amd64

Server:
 Version:         1.13.1
 API version:     1.26 (minimum version 1.12)
 Package version: docker-1.13.1-108.git4ef4b30.el7.x86_64
 Go version:      go1.10.3
 Git commit:      4ef4b30/1.13.1
 Built:           Fri Dec 13 01:48:25 2019
 OS/Arch:         linux/amd64
 Experimental:    false

```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.13.1
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: journald
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Authorization: rhel-push-plugin
Swarm: inactive
Runtimes: docker-runc runc
Default Runtime: docker-runc
Init Binary: /usr/libexec/docker/docker-init-current
containerd version:  (expected: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1)
runc version: e45dd70447fb72ee4e1f6989173aa6c5dd492d87 (expected: 9df8b306d01f59d3a8029be411de015b7304dd8f)
init version: fec3683b971d9c3ef73f284f176672c44b448662 (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 seccomp
  WARNING: You're not using the default seccomp profile
  Profile: /etc/docker/seccomp.json
Kernel Version: 3.10.0-1062.9.1.el7.x86_64
Operating System: Red Hat Enterprise Linux
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 3
CPUs: 2
Total Memory: 3.701 GiB
Name: tebraawx-pr01
ID: OUZX:66KK:TJWB:G6JX:4KMU:HVFB:DXIV:JA4T:K53K:FSO3:5Z2M:EORN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://registry.access.redhat.com/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Registries: registry.access.redhat.com (secure), docker.io (secure), registry.fedoraproject.org (secure), quay.io (secure), registry.centos.org (secure), docker.io (secure)

```

**Comment by thaJeztah** [comment]
Looks like you're running the Red Hat fork of Docker, which has many modifications (some of those may be responsible for this). Note that Docker 1.13.1 reached EOL in 2017, and hasn't been maintained since. Possibly some fixes have been backported by Red Hat, but the upstream 1.13.1 release has known vulnerabilities.

Are you seeing the same problem with the official Docker packages (Docker CE doesn't have a separate repo for RHEL, but you should be able to run the CentOS packages; https://docs.docker.com/install/linux/docker-ce/centos/)


I'm closing this ticket because this is not a build of Docker that's maintained here, but feel free to continue the conversation

**Comment by Wordsgames** [comment]
I followed the docker-ce instructions, I took the latest versions of docker-ce, docker-ce-cli (19.03.5)and containerd.io(1.2.10).

But, I have new error messages that I don't understand on the containerd package

Command: containerd version

INFO[2020-01-30T10:13:41.187676913+01:00] starting containerd                           revision=b34a5c8af56e510852c35414db4c1f4fa6172339 version=1.2.10
INFO[2020-01-30T10:13:41.188523578+01:00] loading plugin "io.containerd.content.v1.content"...  type=io.containerd.content.v1
INFO[2020-01-30T10:13:41.188590348+01:00] loading plugin "io.containerd.snapshotter.v1.btrfs"...  type=io.containerd.snapshotter.v1
WARN[2020-01-30T10:13:41.188954305+01:00] failed to load plugin io.containerd.snapshotter.v1.btrfs  error="path /var/lib/containerd/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter"
INFO[2020-01-30T10:13:41.188995391+01:00] loading plugin "io.containerd.snapshotter.v1.aufs"...  type=io.containerd.snapshotter.v1
WARN[2020-01-30T10:13:41.191863507+01:00] failed to load plugin io.containerd.snapshotter.v1.aufs  error="modprobe aufs failed: "modprobe: FATAL: Module aufs not found.\n": exit status 1"
INFO[2020-01-30T10:13:41.191904888+01:00] loading plugin "io.containerd.snapshotter.v1.native"...  type=io.containerd.snapshotter.v1
INFO[2020-01-30T10:13:41.191961193+01:00] loading plugin "io.containerd.snapshotter.v1.overlayfs"...  type=io.containerd.snapshotter.v1
INFO[2020-01-30T10:13:41.192138790+01:00] loading plugin "io.containerd.snapshotter.v1.zfs"...  type=io.containerd.snapshotter.v1
INFO[2020-01-30T10:13:41.192436334+01:00] skip loading plugin "io.containerd.snapshotter.v1.zfs"...  type=io.containerd.snapshotter.v1
INFO[2020-01-30T10:13:41.192466650+01:00] loading plugin "io.containerd.metadata.v1.bolt"...  type=io.containerd.metadata.v1
WARN[2020-01-30T10:13:41.192503212+01:00] could not use snapshotter btrfs in metadata plugin  error="path /var/lib/containerd/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter"
WARN[2020-01-30T10:13:41.192525586+01:00] could not use snapshotter aufs in metadata plugin  error="modprobe aufs failed: "modprobe: FATAL: Module aufs not found.\n": exit status 1"
WARN[2020-01-30T10:13:41.192547813+01:00] could not use snapshotter zfs in metadata plugin  error="path /var/lib/containerd/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter: skip plugin"

In addition, when I restart the docker service and it starts working but it adds an error

Command: grep error /var/log/messages

Jan 30 10:21:44 tebraawx-pr01 dockerd: time="2020-01-30T10:21:44.582994913+01:00" level=info msg="stopping event stream following graceful shutdown" error="<nil>" module=libcontainerd namespace=moby

then, the same command "Docker run hello-world" that I wanted to carry out and the result changed except the error message.

I have made efforts to unblock the errors but I need your advice.
